src/enc.iced

```diff
@@ -54,12 +60,30 @@ V =
         N            : 15                 #   log_2 of the work factor
         r            : 8                  #   The memory use factor
         p            : 1                  #   the parallelization factor
+    use_twofish      : true               # Whether to use Twofish algorithm
+    hmac_hashes      : [ SHA512, KECCAK ] # Hashes to use for HMACing data
     hmac_key_size    : 768/8              # The size of the key to split over the two HMACs.
     version          : 3                  # for lookups
+  "4" :
+    header           : [ 0x1c94d7de, 4 ]   # The magic #, and also the version #
+    salt_size        : 16                  # 16 bytes of salt for various uses
+    xsalsa20_rev     : false               # XSalsa20 Endian Reverse
+    kdf              :                     # The key derivation...
+      klass          : Scrypt              #   algorithm klass
+      opts           :                     #   ..and options
+        c            : 1                   #   The number of iterations
+        klass        : HMAC_SHA256         #   The HMAC to use as a subroutine
+        N            : 15                  #   log_2 of the work factor
+        r            : 8                   #   The memory use factor
+        p            : 1                   #   the parallelization factor
+    hmac_key_size    : 768/8               # The size of the key to split over the two HMACs.
+    use_twofish      : false               # Whether to use Twofish algorithm
+    hmac_hashes      : [ SHA512, SHA3STD ] # Hashes to use for HMACing data
+    version          : 4                   # for lookups
 
 #========================================================================
 
-exports.CURRENT_VERSION = CURRENT_VERSION = 3
+exports.CURRENT_VERSION = CURRENT_VERSION = 4
```

Comment:
bump, changes default

Reply:
This makes me a little nervous.  Will we need to change code in the Keybase server JS to workaround this. I think we still want to be generating V3 by default for a little while, since we can't expect people to upgrade their clients all at once.

Reply:
No, the version is set in lib_public.
e.g., I pulled in this version into node_modules of kbweb and now the website shows version 4 
![2018-11-30-154843_1385x654_scrot](https://user-images.githubusercontent.com/28712558/49314962-164c6580-f4ba-11e8-87c5-44cb6490bd12.png)

but webclient generates version 3 encoded triplesec's. I added a private key on the website, encrypted it, unpurepacked the encryption, and I get
```
d = unpack (data)
console.log d.body.priv.data.toString('hex')
> 1c94d7de00000003db9759aed3f6fe1ab477952c56541819b636a936b09f2f0087010e6899f81e5056c67bf479b497db2668d0527c2a4d24a959a86dbb9b1114c289e4a284ddb099b60e75d3cdb599978e9c83ff041df6daecf5271f7ebac8977dda44a992b8282094e35c1b4a6235b15685cd530cb43a685cc1f4d878578130a40825646640ad357c0749a8f951f137269d520f817c89e39ca990074a313ac58cda0db871aa163f009eab894fbb1835e1c814523afe0ae9fc07f5ff6cab5437855caad3cfb3774a4146c7197ee3167026c6c6c154d555569809d28adc093951e857556cb141fbe39690f30ed2eb0d540fa7f7938dc3d67f4b7a34051a2b67945d1a6ec68324eca207ad3c1ca16c58267ece6726b286ddec8fa9a80d97ec5dc629cbe7e055b427ec92680f7183fc082cfa992171ffa5d8821680034a16fc870e865d3ed877062b1114cc87f5b0533ae24f19036e174cb6c786b57558e966a3d2f98830cbe405e2569a93791be3e9c9c31096981c9f2c6cfb6a19750a94eaac2deca70991d156d38d4e2cf2e1d10bfd1e321371aa90670369e57c2754793dafffba765373852e52daecbda7d9a2293f7538dc26145c6683370b64a264e20c7974eedd16e6f3e5eedd43bc38964c63de71d8f1d164806f9f1f00e631583538c080cdafd28da0fce403f01333efeb758337624aebe5ef9a7162d85b377b783a5b245df52e148d27fcc7bf58272a71b84a467e01de7837d53997d8fad3fc0457f9a02a39aaa586fae32cea036d2e7d3d9e389187af3cbc986b993df69f487f6f5a87266da8abe66eddd7df5a969855495303c27dd34d782e009761e498179c49b134ca7a97034f4dd009d9b762525dfd1bdef30ac70182ca62801babe2b2dfc58d28887408c30748641f72c140e63ccec98d4e3098fb4ed4e80198fff6d333718ef3144503dfda343366d8830124454b0046d9e16cc30de3755d397900a6a2275f133535f7740bba11b9f2da509ca4415f0c18a534bbb7db1580561720d8e6f7dae796f234cd328532584e305fe5c0d0c3d9401a95bc7abe50cbdea4756e17812b916eed796bb56a1f25f6bf4ee364afb1ff889e36f77a3b1949405cab88b268dc0d188bfa81348cc05ef6adb79376cc95ea8f9a00e823d765c9c27579d1dae3049064f5471b277147bed40c2e65a62b5ef5df6db71b5db73ff8efece563d60cbaf7f6ec470501e2a5c7fccb8c0da32979aff6253ed36f6607e5b4128f321297c9e136cbd59022c0073a9afda2befaceeffbde6edf1f00da239822f31a5ae403e04b6d1801fa0b146764f4ef02622f2e8bea08ba245bd9aebfc2e820d3b46b3592e2f512774868273c357735c8e6262bebdddc7e442861190ae2257722aa1201d992d4e06e1741aeef5a4568cd43912365abb255b2cefa1ef36d168a9b0aa0ff0cfd66ccc50ab4966481b262b30ca3414ca7ae1b84e7006ec090209f1051348e65e8d367acd00eea25a15071af3d412d80baf9cf3fede6e6b3f77aa5c787db7c04ee58212e26b9cdaaf617df2a25ab2cef0d3f4238e1d834ccbdccea3215cb4453eadd1d357e1bf93ac899d05d0cfd9c5d0c24662ea9b2413478bd4e44cc5482500949d65be79ff6608ea8702e3963d6bb161ee58045eae00690f48a371ee95631da448692e48bcd734db5a246611409784408cf6cef3ab1f6d7420fc4b64c5fcfd596812a7e1951d2a1c6be55a58aa867acab865937e811d30c9551901b745868227167f58a3003439c1a83d1d4b88971c08e5ff68a7e88900c6b3f0a0317882adfbe23d342516e53a1d81753b0c92eda8373db02ca04628ed3dff51a6b9efb3cd02749bd81dcde2e3a5e7c301913ff27c8f04606cff22eae9fa309b18c6cbbbe20b7439c6ae4bdcd248e498d0e0bb6be91f071d4f6817dde831fbe169fe46a98f56e37d9864710a277381413f8c3ff8ebab1d4279b8caa58fbe3ee6e12fbf44b9c575c103ca3b2a908a5369bcf03f57ac15c9c4d3a5a200d1b002aa4b1aac8a051a0dd105987df75b7b2370409f7dde36436693bd303726363be95ac54c69d966b6132b4a2f022df43a4e341e8b39505329a07bdebe690f42f02ab1a6f19519aa2f5246eb289580d8652152b8edafe1f095e770f9426fbb7a69c14ed127972528172d126488c0c103ff21d91d06b211af4fef286dbd6dfa41aca7b07a198c28fd5196bce7b4e5b2e823f7edb6cf3f9585d1bdbfec76efc93055c7b551bbb438b158b266d9023f8a87be91b1396fd02864d0a377963864aabebab7e6269472646b5247cef83ee4d06d3db9d520dbba5056f661ca176c440ea88645f26f2b2ddab1d2e1b2fce64d838cc3cbbbaad26c728ea9a76a0014d965c4e19b9837adfa259c05b0784bec33bdf986239f9feeadeb9497c6ee9e7751c9b645c179e10675ba58c31ac8209a308d60a22941769e0c57b899bde73d957576c9ba205ff5a68ba3cb9882f7cbe6d65787d6129c7c036c767260c1398ddee61f0cc557cffa7bb0de8853918780befde6a3601a6ec89339fca058702cdbfd332ff4fb4a9b0f4aa14d346fb8f1e45cb8622899c3399aa50d4b49a9748f22700b8780414d1089bc9356a6616bdf25aabf6c80ea186d09f49e7e81297096df4950d90addc309af72af8ac6124e1da31e728a1a83b41e39c2f03123c2b181c4b72262b11595b72498c2a8aa376abb25cd828c2cec79d8e1395711cefa69ae304065ed2d195909d9ded8b5b5cbbaa42f3565435dc5a1b5adead0cebdcadba715798f9b6fb842f7c4e29228b7394398acb7981fd349b4b95f929de92e7b33f6d6bd96af657073c4f3375d6f1ea7f616dff30d2b75cb964b4c432ab8ef451ad90805ca17cba9ca089ddb17cdb38a9b021436209695bf75b8701f3ddbf93c567fcefa068e428445c24121f638b5f7ede81829084e7b3daa44991fe4d3c2d8c7f61de8a2112a33552028234cf0122f0f0776bf9247bd43b2ed9ccbddf1e1e0118ec713fc7e42ffb571978ba68ec6acb2b01ed31f631d294e7b9777a5ac0d743f20218f3eb5aea6213ccf27c3ad453cfc5cdac2bfaba05cba5685edcf098e0e5e3b6bd98d4887246f749dbbf253b260c4f7ccd8d1a6519b1a175f97688a3f3c7a9af020aab4b54da63144b7376db0d0c99e5daa4f158fa5fb137167e81e17ec54b6f390b615e32fcfd458655e3ba03c2448b8205b9aab4248733f46c00e983fc34372895735d757610d1c259b6c7523ccfdc73f3942a35504a15d58f910ffe7d93b3613f2a36cabeff5c118ad2f846f5625ed2bb36fb1fe2185a382e9114316bea977a0c9fe0d27dbea100dd420027b430e5e06715d2d6b4de2ff2a8cb28e9c076ae22fd14e6acb749a195b541442e35e6dfbab61d78f1072336bdfba774df2bf54361a950bc2a4e9a093ca88dc7480a296f77c79453b7d31479cacc77dcf52a46a9f6ceec4b136b80edf0bf4b849a8c7c69007ca978fa925fc9d7b6985a758164144ccca086459d9d435807b489ea50618e5538f4c89d55513848aaeb05b0900917e549c7a61280b7304d6d9eaf9029b0688e0c12a3e3827d57d606e04e26d4e76171f172e0b3cd53e096169643655c1f25fba508535f65c2ca117761eaf12697852f13df8fffc7c85ba4c507634a3317ad56ab92991b90ce05ca1daca8566f916f77967b5d3ac57b3f13e08783ee14e3f68cab877f08b107c2c31ad445daf89d24be7d7628482d43ec3c0d1eb62f28fd0841c3e44a88d967416d9863366165748cb1f7e7d0851f0192fe12d890a1ac032a1d15a3cb6ecc25bbed6c6641aa3e607a8fad0a6c94984c00d9b7ddcb680fbb3d62896d8b7907de30c9a1e8336d43fd0ed350d72b219bf4cbd15a10563866d2c730594eeb9e56a11dbbb8c7e09b4cfcc629dda32982dbd3b5a2917293b8b72957b32291ebef425d7e44316e92fd09c64854b3296f5159d7968f8d35656395e5714d3f01a07f05278238b906d342768bf0989bf026c4f9e8528857dd8587f04ea098fec35afe2e20c6f0ac975d14aad10090e7a3b8cb03dcb5383f15f660f1add8469cb18d375a0dcb18337d91e471c0c9165e77394b082106b062277295397400469e2ee3528b4018cafad1c0522d94302a747c6979346d535c8386e928d80f745fa30d43d55fc2366ea654d70f25adbac502ee2af4df000ae11c03210a1e4668e23d4fc14f71c3f047399ef4c6e7f3152f764447de49fee7e21c324fa90639b4e35a5570532719b5f3a34d1cce2420eb687bbce6e47a2b16468cfd67ca1decb0848b551ebf2903f544f8147c8326bdbf02541763a79b866cf6a4a092d9c409b55b662441d488591be045d5c3ec2d1d968344e649a9849a9b9d8d216c0b1a8eda5d767839d150d23069ce58de4c6a16569ba8f81e17622d460f67e6a4b468c94165e059a354c3827c4250e7aab004693e689b92c8b70454782c6155ccf7bc756943664323ef50eb0f912c12ebc7fa30a189ef003d09386e6bb10aaa1dd501de61c40e36ce83f0c18c5f01e98b2e52a4e919426955a0a828ef2e257a194b347d6e61a386e7e28ea9bad2b2e2208dc144a2124984716f91181a640135b30cd04c2e2e0ce66565b7781c5023575f5e1f0567302b7a6f11b929907357a93ab041e9ee8b048e61d0844fe09f1a85257564cfaeeda55677bbb6c0625684f8a4f356196601517b09a2b8742e91d86cadc22340c08146801c5f615800fb7906f1c9da9e9d9e932c2072052b742bd722807598f4778784ba84540b868bf4f912ae6cf47f31b4273bf1d3dc68ecd6fe805c21162b15f6ab3e43854734bae9e505e78721c4a1736c660ef21b2a5a3b06af7ef742e1c65a83283b67643858c555d44e0ef37b7e6576f4891c32f907f8d56215195b2d32225f4c1de7cb4bd36493b916a41222e8a77c9181b8b98144dcbf6530839d82d468dcb27a57618aceae3a0ca0636c6dbc5a34d84faf95b37bc23bb84ef766aa9f3fe961fc0dade013a971ef2fd54ce8d962e81a4e1b1eca700e3b097c2cd05cdcd38c31b5098be4541381dcd5d329bd857e1d097aa6f73f7f08c548bab199acced1a97679813f3f6bfbd99a3c1dd3537eff1d918f913527e98bd5cdc8d9d356aa0a7304d372c6fcb9a4296c7a6d62592f482cc9b6796b62b75de42e00e7b9789e737256e111c24de2fc787154bd8de08417c0d3548e786d3702d2159149a26a59e03a93a39796e9702c70d33d23c9b81f579545d98c49b85593e0b19c5f0d1ed96c7ac289481c3dcba423c1b4fe7fcbc8e08e1c4ff60d21ee54cf56664fa7567f496f7cfa1f19ed76b989131816fdf9aa2bbe706c2ae070bd8e96b5fc6670a0239bee532b906feddd7987b520364cdd58d024a064ac94f8d0e8aae789e45d1d125a73265d3e72873e14be92568ef67121e6efbc8a19354d0fe16756ccfc3a1a2db6eaf38294e0bd13887c22c99a72616a5b44a9040a66e29c9a1bb33de59de008ec78365e8de40ad687a3b8df17075381912881a6ed4f46b4ee7f36e6931869d0cb3ebb04de7bc931c5adaf502e8b9cc9997a0b481942bd0a0e42105cc6b4f7f660a6313ea8b967924a701d441fb8b50ec9cf2c8cb0b83ab1648ed1759e68efb6ca6ffa938851f59060158e08ca9a34355278e1b393176cd3f1e62a86e01ac67284c8ca68a7a1589e124873106e2c63bf13e3e458b290bab2b00fad43469e30c0af7f6da2601e8155318d4a168043560272efd1da8730bbef52eb93f4b44961b4c3bb63d4ed8a0919caae0a37ca682a2a526942771b4ad6ccd240d27c748cb9649debfad724b6cd5e62bf3b1ef142c764c32a7d09efb00e98c8404027bf9e9f1aad3c29ca2ad5fefab34490c15973d919405d8b8d1a4b63ca6e4adf4c8a7e99f4c2ffefce0c5abf1307ae1b4a75a8a416edd9108d489d9d1a07dc63c7d3b88674c04fab23d8f994e4554fd9006ad0ca3ee0fc9004994e452c7995bbb8476ca8519e9e8947b70b96b24726fcaff53726e8ba4cc29878cb006c927b8205adc445ed8933c00ccb7e6a0e574e1b88205fab98f13bd7653004d1f34154ac99cf0765d06ccbf68d7b03dd7ddb0800af8393f8ceff7014109c5bab1406209427dca78f6388ed5f5da9d774339a421b0eb06512bf4e9d72f3d97eefcee2218c4f998289224c48fad4a8eedb2db2f33be3bbacb583e450ecc9a45d72ed120acd1e2d390e9addd555d5470e5dd52326ac0f89615d57b007378bd8ee2d4104327c43aa66ed13581584d8ab495fa3a7d8b42a41cbab6729854099491468404c6b5a3512917cbf8fafe7f84384802ee2b3c57591745390fff5283b728c5d12e323c96cf5d0105b8b7d5262f1a04d4559c1e277a3b166eee60c7e9a121058595c32dc7117685cff31158535804333637a59d1777340a879c843ba374cae26788d30856d9f286c1a98a6b8eedff9d66a9ebbaecfd49f8965a46449cb3c455dd1803035b58c53a93ada4584b91e7ad96e2464a79dfc92ec949c1a2f4b23f3824f3d9e1767e4b6e05139997fe35c1705adb7333dee2f4329fa0c52b241807aaad17d38e21b56eae6b03c35a4a67948eb5341c5227166e9037733ae3d550a444f5eec4e89300f17f14396122fdd0e1efe0815384bbbaaed009f4c0308834181f0e80cbef866fc0926b8117edf92828d90660e41cd6df1298bea0b7f5b885019de2c5da8ddf0e860fc1a921bd6cd5839c381e1b4a6eeba649117c2d8e3db3fc0011e7d0f896d1d0de67afe5e1cc9fbac8db90c2fe36ced989e7ff55e97f77336359ca4c40f64ce115dca63930454087c26c73daaa3997ba1789906fa7a45741e1446380d5cbb638d5d341b91090f3a780fcd1ae78b78d6e157dee33644960cdcd25e1a72a93345576c133e05373e4bb61dd75d65ee29c94a67f7c6ef5a393ca6f65890f4ca0ce8cfe0d43c5be1d2f51e45c9051e8a21ac085766836349b3ba9f964f4cb6c481c3901b4f2c2dcfa9d7917d8593d1ba27e62ea494e58598aa1364363e6c6ac3c069151826426541f2b91930c5e8da1f453955961fde96ce86b4c0a1f5ef22ad10cc074b4c404468ef2b18d6141155f779c2f12fa1e6cd60933ce7a3b11174fda4be9f96f5c8694ae21a144b162bf48625ebe3b2afb4735b16ba12849f86f7e229d700dbe2f7cb707b7a07a9f108566dcde5b67534c45fd42bdd4d7b61ebdf8c036493648ffe8d688fecb887282956755044742df38416c3d3590b7f5c6f0415b6e1cc9eb3e81ac06e8ec61bc9c2fa7c5eb1d1c6010dd13cea08df29527637256c6b6d400c26bb52a3b2ad1423f985262734aa6e02c1a4030d717d5bbc12d1f2b012d43e7b0e4eba3eb2752da82d5a1e6f4c8019112a06b25b335a6691992ce78b91b69f2ab1a03ebc9d0e3e561afba97765c1da2d40d5771f0f5ab06298148c99eba81a4aff0e946f963bfa1901e062698014bb907891400014cbc001a2b9958bb67e26739032f29b539816e05dad567a2530435062079f728a1d5c04e97b9b3d348dc81c8f788c39bc495bfbafa38b7936e80bd844fc514daa387ef64f50fc5b5fffdc19f820e5c25a3bacc05296dd4c153c442cc05320954161d6b3f388a87c15d5628590547ff44f766cc0ba2fc870ea1cb818a6522990c49a7f6fcc923322c3910f27ff053af2d90a2db242b810703387856376699a655e884847795c9d98b95af43f1c36dc0c891cabc1ef9cfa5ee678a00f9c3e3a45b24f755c4df0d4f8f0d82be5a377ad0a0b75b651e4a837cb348005cc38fcdd71d96447881cc2d1cf4b0e5124f8956e6065e63ea22b62b9b09dead5073c682ccfd96226c5b4a83cc3aa5ee93d1e77d44b549e89a89d8a7c0cfe88be90a80b729b1dabc03973b6551b2864fd659c313fb3940810ae46f01217fbfe2a9230b169549a387d77f71a704e23bcfeb7e22cabb84c896a3be0735e5d67da1f7f6edf6526ad2ab333282a904655821854941ffb8c3f5513c48bed15ca5b9fb2111141c562df476dc94f8413896cd27ce64258cd2e9d700ed6f8ebee84978d08036cbaa304c2592f72acefdf25e3fab55ab97c77d319a898118cc3118fd3da557c2cab9ce788026447048de21aa3760f4797743ad90e9ca2b82634bc96701d3f896554613c1f311bacdd28b7b2bd4bb9151ccc93a04fe16ab9f6ca3706a24bd6a377f4007b1c83e1a1bfc4023d06913a7b41bf74c1c0a8148040cc9cfed716a79e12f8eaa03a92415fe71d0a8415ae7e458471d5c43b494486b8e7ba3082840f4bef4f9c3f3bd698a19bd13de59292eb7a5e684fee68db749021431098cb62e04f032ad5eaa84e20e566245485430efb872608562081c21f74cb49e32437e48c13c1be47d4f04ac8d3d65c1638a32000721167d2c1eae26471c41cf6997d8e4eced66cccf8569ea077e2ff7932d4952d85cd57207738f4275bd512ba73fbfb6a68048c3b7a43629b66e5495b2f7452f12fe3f11108d9443ecffc3d26a79f49dad2fa192a3ff136431daebf65a3d42275ce89190cbf949cac6e69079c30445a8a988b2ad3272a86f5bcf4676a48cb1a7187aea44ac3128ca9925bf85a0e6f59fc1bb1a35bf9cc713f0574abfb8813ffd28e01cafeab3498136a5b18c91bc0d2cd48e4aeb5114fc34711ed307843cfb15b2bc025a52fcf220e38161839addd767122487e73f05c
```

which is version 3, and that I was able to decode with the new python code (not possible before due to keccak issue) and also go.